lib/matplotlib/backends/qt_editor/figureoptions.py

```diff
@@ -165,37 +177,35 @@ def prepare_data(d, init):
 
     def apply_callback(data):
         """A callback to apply changes."""
-        orig_xlim = axes.get_xlim()
-        orig_ylim = axes.get_ylim()
+        orig_limits = {
+            axis: getattr(axes, f"get_{axis}lim")()
+            for axis in axis_map.keys()
+        }
 
         general = data.pop(0)
         curves = data.pop(0) if has_curve else []
         mappables = data.pop(0) if has_sm else []
         if data:
             raise ValueError("Unexpected field")
 
-        # Set / General
-        (title, xmin, xmax, xlabel, xscale, ymin, ymax, ylabel, yscale,
-         generate_legend) = general
+        title = general.pop(0)
+        axes.set_title(title)
+        generate_legend = general.pop()
 
-        if axes.get_xscale() != xscale:
-            axes.set_xscale(xscale)
-        if axes.get_yscale() != yscale:
-            axes.set_yscale(yscale)
+        for i, axis in enumerate(axis_map.keys()):
+            ax = getattr(axes, f"{axis}axis")
```

Comment:
```suggestion
        for i, (axis, ax) in enumerate(axis_map.keys()):
```
I think should be equivalent.

Reply:
Yes, that's much better. Thanks for the suggestion!